### ISSUE TITLE
Figure out a noise protocol udp-based transport

### ISSUE THREAD

**Issue by iglesiasbrandon** [storj/uplink#116]
We should be able to use the IK pattern (https://noiseexplorer.com/patterns/IK/) for making 0-RTT enabled RPCs to nodes. The "long term static key" for uplinks is actually the key they generate at process start, and the satellite should be communicating the static keys for the storagenode.

How should stuff be layered? For example, wireguard is basically this but it encapsulates IP packets (see https://www.wireguard.com/protocol/ and https://www.wireguard.com/papers/wireguard.pdf). This is a great approach, but we can't just slap TCP on top of it because we still have the 3-way handshake. 

One fundamental question is do we do the noise below the reliability/stream layer, or above the reliability stream layer? Below has the advantage that noise is sorta packet oriented already. Above has the advantage that we don't have to worry about nonce ordering/reuse issues. Above has the downside that it may make 0-RTT more complicated maybe?

Basically, we have a packet-based udp connection as input, and we need a 0-RTT enabled streaming connection as output. How do?

**Deliverable**: A high-level design doc outlining the protocol/framing/etc.

Things to consider (just a starting point hehe) : 
- how does it get encryption 
- how is the reliability
- flow control congestion control
- the kernel has UDP interfaces that allow receiving multiple UDP packets at once
- we may want to allow multiple streams over the same session
- we may want to make the least number of copies of the data

**Comment by iglesiasbrandon** [comment]
going to close this issue for now until we decide if in fact we are going to use UDP or TCP fastopen.
